EpiAware/src/EpiAwareUtils/accumulate_scan.jl

```diff
@@ -0,0 +1,58 @@
+@doc raw"
+    Apply the `accumulate` function to the `AbstractAccumulationStep` object.
+    This is effectively a optimised version of a for loop that applies the
+    `AbstractAccumulationStep` object to the input data in a single pass.
+
+    # Arguments
+    - `acc_step::AbstractAccumulationStep: The accumulation step function.
+    - `initial_state`: The initial state of the accumulation.
+    - `ϵ_t::AbstractVector{<:Real}`: The input data.
+
+    # Returns
+    - `state::AbstractVector{<:Real}`: The accumulated state as returned by the
+    `get_state` function from the output of the `accumulate` function.
+
+    # Examples
+    ```julia
+    using EpiAware
+    struct TestStep <: AbstractAccumulationStep
+        a::Float64
+    end
+
+    function (step::TestStep)(state, ϵ)
+        new_state = step.a * ϵ
+        return new_state
+    end
+
+    acc_step = TestStep(0.5)
+    initial_state = zeros(3)
+
+    accumulate_scan(acc_step, initial_state, [1.0, 2.0, 3.0])
+
+    function get_state(acc_step::TestStep, initial_state, state)
+        return state
+    end
+
+    accumulate_scan(acc_step, initial_state, [1.0, 2.0, 3.0])
+    ```
+"
+function accumulate_scan(acc_step::AbstractAccumulationStep, initial_state, ϵ_t)
```

Comment:
Maybe slightly clunky to swap args order compared to `accumulate`?

Reply:
I just did this to stick with what wee had (in scan) but I agree should reverse.

Reply:
I just looked at this and going to lightly push back as doing it this way would mean the step functions and the input are in a different order (look at the example of `accumulate_scan` which felt a bit weird to me on reading. Happy to switch if you look at that and disagree